Examples/Run/Reconstruction/MLBasedTrackFinding/TrackFindingMLBasedExample.cpp

```diff
@@ -0,0 +1,282 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/ExaTrkX/ExaTrkXTrackFinding.hpp"
+#include "ActsExamples/Digitization/DigitizationOptions.hpp"
+#include "ActsExamples/Framework/RandomNumbers.hpp"
+#include "ActsExamples/Framework/Sequencer.hpp"
+#include "ActsExamples/GenericDetector/GenericDetector.hpp"
+#include "ActsExamples/Geometry/CommonGeometry.hpp"
+#include "ActsExamples/Io/Csv/CsvOptionsReader.hpp"
+#include "ActsExamples/Io/Csv/CsvParticleReader.hpp"  // for evaluating performance
+#include "ActsExamples/Io/Csv/CsvSimHitReader.hpp"
+#include "ActsExamples/Io/Json/JsonDigitizationConfig.hpp"  // to read digi config
+#include "ActsExamples/Io/Performance/CKFPerformanceWriter.hpp"
+#include "ActsExamples/Io/Performance/TrackFinderPerformanceWriter.hpp"
+#include "ActsExamples/MagneticField/MagneticFieldOptions.hpp"
+#include "ActsExamples/Options/CommonOptions.hpp"
+#include "ActsExamples/TrackFinding/SpacePointMaker.hpp"
+#include "ActsExamples/TrackFinding/TrackParamsEstimationAlgorithm.hpp"
+#include "ActsExamples/TrackFindingMLBased/TrackFindingMLBasedAlgorithm.hpp"
+#include "ActsExamples/TrackFitting/SurfaceSortingAlgorithm.hpp"
+#include "ActsExamples/TrackFitting/TrackFittingAlgorithm.hpp"
+#include "ActsExamples/TrackFitting/TrackFittingOptions.hpp"
+#include "ActsExamples/TruthTracking/TruthSeedSelector.hpp"  // for evaluating performance
+#include "ActsExamples/Utilities/Options.hpp"
+#include "ActsExamples/Utilities/Paths.hpp"
+#include <Acts/Utilities/Logger.hpp>
+
+#include <functional>
+#include <iostream>
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+
+using namespace Acts::UnitLiterals;
+using namespace ActsExamples;
+
+ActsExamples::CsvSimHitReader::Config setupSimHitReading(
+    const ActsExamples::Options::Variables& vars,
+    ActsExamples::Sequencer& sequencer) {
+  // Read some standard options
+  auto logLevel = Options::readLogLevel(vars);
+
+  // Read truth hits from CSV files
+  auto simHitReaderCfg = Options::readCsvSimHitReaderConfig(vars);
+  simHitReaderCfg.inputStem = "hits";
+  simHitReaderCfg.outputSimHits = "hits";
+  sequencer.addReader(
+      std::make_shared<CsvSimHitReader>(simHitReaderCfg, logLevel));
+
+  return simHitReaderCfg;
+}
+
+ActsExamples::CsvParticleReader::Config setupParticleReading(
```

Comment:
These helper functions don't already exist anywhere else?

Reply:
I just took the example binary as it was by @xju2 and did just a few adjustements. Probably there can be done some more refactoring, but since we plan to phase out the examples binaries anyways, I thought it might be okay to leave it like this. But of cource I could do it still...